packages/storage/amplify_storage_s3_dart/lib/src/model/s3_copy_plugin_options.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+part 's3_copy_plugin_options.g.dart';
+
+// {@template storage.amplify_storage_s3.copy_plugin_options}
+/// The configurable parameters invoking the Storage S3 plugin `copy` API.
+/// {@endtemplate}
+@zAmplifySerializable
+class S3CopyPluginOptions extends StorageCopyPluginOptions {
+  /// {@macro storage.amplify_storage_s3.copy_plugin_options}
+  const S3CopyPluginOptions({
+    this.getProperties = false,
+  });
```

Comment:
These need `fromJson` factories to avoid triggering some unused warnings in the generated code. Or you can replace `zAmplifySerializable` with `JsonSerializable(createFactory: false)`

Reply:
were do you see the warnings? I ran `dart analyze` and `dart format` and did not get any warnings/errors

Reply:
They will show up when published to `pub.dev` which uses a different set of lint rules, e.g. https://pub.dev/packages/amplify_auth_cognito_dart/score

Reply:
I see, I will fix this in a sep PR. want to merge sooner than later to avoid merge conflict. 
btw what do you think of using same lint rules, https://pub.dev/packages/lints, as these have negative impact on pub points?